try.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2022, Joe Tsai. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE.md file.
+
+// Package try emulates aspects of the ill-fated "try" proposal using generics.
+// See https://golang.org/issue/32437 for inspiration.
+//
+// Example usage:
+//
+//	func Fizz(...) (..., err error) {
+//		defer try.Catch(&err, func() {
+//			if err == io.EOF {
+//				err = io.ErrUnexpectedEOF
+//			}
+//		})
+//		... := try.E2(Buzz(...))
+//		return ..., nil
+//	}
+//
+// This package is not intended for production critical code as quick and easy
+// error handling can occlude critical error handling logic.
+// Rather, it is intended for short Go programs and unit tests where
+// development speed is a greater priority than reliability.
+//
```

Comment:
I've been using this in package main, and leaving off the defer. Might be worth mentioning here.

On that front, it'd be nice to provide a catch variants that doesn't require a `*error` but instead passes err to fn. And also provide adapters to testing.TB and log.Fatal. Then you could do:

```go
func TestFoo(t *testing.T) {
  defer try.TB(t)
  // ...
}
```

where `CatchTB` will call t.Fatal with any error that occurs.

And

```go
func main() {
  defer try.Fatal()
  // ...
```

where `Fatal` will call log.Fatal with any error that occurs.

Maybe even a `Log` that calls `log.Print` with any error that occurs.

Reply:
Also, for all of these uses (testing, main, logging), you really, really want line numbers. (See my implementation at https://github.com/frankban/quicktest/issues/138#issuecomment-1114014410)

Reply:
I like the additional functions. I'd have to think about the semantics a little more, but those can be subsequent API additions.